### DISCUSSION TITLE
Uses of local variables within a function

### DISCUSSION THREAD

**Discussion by Cyleigh** [NationalSecurityAgency/ghidra#3172] [Q&A]
Is there a way to get all the PcodeOps within a given function for a given local variable?  So far I can find the HighSymbol given the function and the name, but I want to then grab all the uses of that variable?

```
DecompileResults res = decomplib.decompileFunction(f, 200, monitor);
if (res.decompileCompleted())
{
    HighFunction highFunc = res.getHighFunction();
    LocalSymbolMap localMap = highFunc.getLocalSymbolMap();
    Iterator<HighSymbol> localSymbols = localMap.getSymbols();
						
    HighSymbol localSymbol = null;
    while (localSymbols.hasNext())
    {
      HighSymbol current = localSymbols.next();
      if (current.getName().equals(theName)) { 
       localSymbol = current;
	break;
      }
  }
}

**Comment by ghidracadabra** [comment]
HighSymbol.getHighVariable() returns the high variable associated with the symbol.  HighVariable.getInstances() returns the varnodes comprising the high variable.  The methods Varnode.getDef() and Varnode.getDescendants() might then be of interest.

Also, make sure that you are using the default simplification style for decomplib (which is "decompile").

**Reply by astrelsky** [reply]
What is the reason behind using the default simplification style?

**Reply by ghidracadabra** [reply]
High variables are only created if the "decompile" simplification style is used. They're not created if you use "normalize", which some of the example scripts do (such as GraphAST.java).

**Reply by astrelsky** [reply]
> High variables are only created if the "decompile" simplification style is used. They're not created if you use "normalize", which some of the example scripts do (such as GraphAST.java).

I see; thank you. I should have just looked at the docs but that information is good to stash away somewhere.

```
The current predefined analysis class are:

"decompile" - this is the default, and performs all analysis steps suitable for producing C code.
"normalize" - omits type recovery from the analysis and some of the final clean-up steps involved in making valid C code. It is suitable for creating normalized pcode syntax trees of the dataflow.
"firstpass" - does no analysis, but produces an unmodified syntax tree of the dataflow from the
"register" - does ???.
"paramid" - does required amount of decompilation followed by analysis steps that send parameter measure information for parameter id analysis. raw pcode.
```
